pkg/operator/workload_controller_openshiftapiserver_v311_00_test.go

```diff
@@ -0,0 +1,166 @@
+			configObservedGeneration:    100,
+			expectedProgressingStatus:   operatorv1.ConditionTrue,
+		},
+		{
+			name:                        "ConfigAndDaemonSetGenerationMismatch",
+			daemonSetGeneration:         100,
+			daemonSetObservedGeneration: 100,
+			configGeneration:            101,
+			configObservedGeneration:    101,
+			expectedProgressingStatus:   operatorv1.ConditionFalse,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+
+			kubeClient := fake.NewSimpleClientset(
+				&corev1.Secret{ObjectMeta: metav1.ObjectMeta{Name: "serving-cert", Namespace: "openshift-apiserver"}},
+				&corev1.Secret{ObjectMeta: metav1.ObjectMeta{Name: "etcd-client", Namespace: "kube-system"}},
+				&appsv1.DaemonSet{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:       "apiserver",
+						Namespace:  "openshift-apiserver",
+						Generation: tc.daemonSetGeneration,
+					},
+					Status: appsv1.DaemonSetStatus{
+						NumberAvailable:    100,
+						ObservedGeneration: tc.daemonSetObservedGeneration,
+					},
+				})
+
+			operatorConfig := &openshiftapiserveroperator.OpenShiftAPIServerOperatorConfig{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:       "instance",
+					Generation: tc.configGeneration,
+				},
+				Spec: openshiftapiserveroperator.OpenShiftAPIServerOperatorConfigSpec{
+					OperatorSpec: operatorv1.OperatorSpec{},
+				},
+				Status: openshiftapiserveroperator.OpenShiftAPIServerOperatorConfigStatus{
+					OperatorStatus: operatorv1.OperatorStatus{
+						ObservedGeneration: tc.configObservedGeneration,
+					},
+				},
+			}
+			apiServiceOperatorClient := operatorfake.NewSimpleClientset(operatorConfig)
+			openshiftConfigClient := configfake.NewSimpleClientset()
+
+			operator := OpenShiftAPIServerOperator{
+				kubeClient:            kubeClient,
+				eventRecorder:         &mockRecorder{t},
+				operatorConfigClient:  apiServiceOperatorClient.Openshiftapiserver(),
+				openshiftConfigClient: openshiftConfigClient.ConfigV1(),
+			}
+
+			assertNotPanic(t, func() {
+				syncOpenShiftAPIServer_v311_00_to_latest(operator, operatorConfig)
+			})
+
+			result, err := apiServiceOperatorClient.Openshiftapiserver().OpenShiftAPIServerOperatorConfigs().Get("instance", metav1.GetOptions{})
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			if !operatorv1helpers.IsOperatorConditionPresentAndEqual(result.Status.Conditions, operatorv1.OperatorStatusTypeProgressing, tc.expectedProgressingStatus) {
+				condition := operatorv1helpers.FindOperatorCondition(result.Status.Conditions, operatorv1.OperatorStatusTypeProgressing)
+				t.Errorf("%v: expected status == %v, actual status == %v", operatorv1.OperatorStatusTypeProgressing, tc.expectedProgressingStatus, condition.Status)
+			}
+
+		})
+	}
+
+}
+
+func assertNotPanic(t *testing.T, f func()) {
+	defer func() {
+		if r := recover(); r != nil {
+			t.Errorf("Panic: %v", r)
+			t.Error(string(debug.Stack()))
+		}
+	}()
+	f()
+}
+
+type mockRecorder struct {
```

Comment:
Move this to a subpackage under spot where the in memory one is.  `TestingEventRecorder` and `NewTestingEventRecorder`

Reply:
Opened issue and will use `InMemoryRecorder` for now.